### DISCUSSION TITLE
Highlighting and auto completion does not seem to work

### DISCUSSION THREAD

**Discussion by kanodiaayush** [nvim-neorg/neorg#77] [Q&A] [answered]
highlighting and auto completion does not seem to work in neorg as suggested towards the end of the README

I am not getting suggestions to auto complete check box styles for example, only very basic auto complete.

Further, no highlighting. What am I missing? I followed all instructions on the README, and installed and configured TreeSitter. It looks like it is working (highlights) for eg for python scripts though

.norg snap: 
![image](https://user-images.githubusercontent.com/1688433/131047164-d5729fed-d594-4d75-948a-2dab34126d5f.png)
.py script snap
![image](https://user-images.githubusercontent.com/1688433/131047242-ffe969ef-56b1-4e92-83a2-0c6e47bfa75c.png)

Please let me know if i need to add config information from init.vim . I have nvim 0.5

**Comment by danymat** [comment]
About highlighting, did you put the language name after the `@code`?
Like so :
```
@code py 
print("hey")
@end
```

**Reply by kanodiaayush** [reply]
Yes I did; that did not help.

**Comment by mrossinek** [comment, marked as answer]
Can you double check that you have configured the `norg` treesitter parser according to the README as follows?
```lua
local parser_configs = require('nvim-treesitter.parsers').get_parser_configs()

parser_configs.norg = {
    install_info = {
        url = "https://github.com/vhyrro/tree-sitter-norg",
        files = { "src/parser.c", "src/scanner.cc" },
        branch = "main"
    },
}
```
This is a lua snippet so if you want to place this inside a `.vim` file you need to place it in a `lua << EOF ... EOF` block.

**Reply by kanodiaayush** [reply]
I did do this (and I submitted a PR to add this comment about `lua << EOF .. EOF`) , so this is not it. Thank you.

**Reply by mrossinek** [reply]
Did you run `:TSUpdate` since then?

**Reply by vhyrro** [reply]
It may also potentially be due to loading orders. Make sure neorg loads after nvim-treesitter by executing the neorg setup function after nvim-treesitter's if you're not already :)

**Reply by kanodiaayush** [reply]
Yes I did run `:TSUpdate` @mrossinek . I do have treesitter and norg setups in that order (also sourcing plugins in that order). But that does not seem to work either. I am likely making some stupid mistake.

**Reply by kanodiaayush** [reply]
[init.vim](https://github.com/kanodiaayush/.cfg/blob/master/.config/nvim/init.vim)

You could look at the entire init.vim but don't want to bother you too much too @vhyrro @mrossinek . Thanks.

**Reply by mrossinek** [reply]
I do not see the snippet which I shared above in your `init.vim`. You have to add that and run `:TSUpdate norg` afterwards otherwise this cannot work.

**Reply by kanodiaayush** [reply]
oh thank you! I did not understand from the readme that that goes in every time (or is put somewhere in init.vim) - because it said that you can put it in some file and I guess i understood it as you need to run it just once. This worked!! Thank you very much!
